### PR TITLE
Adding the FOXSI PSF for any position to the IDL repository

### PR DESCRIPTION
**Pull request by aringlis**
Work in progress - not to be merged yet. This is the IDL version of the functionality added in #20

### CODE REVIEW

**Comment by ehsteve** [comment]
Fixed my PR so you should update yours I think.

**Comment by LinErinG** [comment]
@aringlis @ehsteve I'm missing psf_parameters.txt, so the code doesn't run for me.  Should I already have that somewhere?

**Comment by ayshih** [comment]
@LinErinG @aringlis I'm currently fixing up this PR, so you might want to wait until I'm done.  But to answer your question, it's from Steve's PR.

**Comment by ayshih** [comment]
Andy needs to merge the PR I just submitted to his branch.  Here's an example (with oversampling set to 3):

screen shot 2015-11-11 at 3 30 29 am

I have not yet cross-checked this against the Python side, the IDL side looks to be doing the right thing.

**Comment by aringlis** [comment]
I just merged in @ayshih's updates.

**Comment by aringlis** [comment]
Removed the WIP label but it would be good for someone else to test this out.

File: idl/foxsi-smex-setup-script.pro [hunk 1]
```diff
@@ -9,7 +9,8 @@ CD, Current=fs_location
 ; with foxsi_
 
 foxsi_root_path = fs_location
-foxsi_data_path = strmid(foxsi_root_path, 0, strlen(foxsi_root_path)-3) + 'data/'
+;foxsi_data_path = strmid(foxsi_root_path, 0, strlen(foxsi_root_path)-3) + 'data/'
+foxsi_data_path = foxsi_root_path + '/data/'
```
Comment (ayshih) [inline idl/foxsi-smex-setup-script.pro]: @aringlis you need to revert this change.  The weird (unexplained) code is to strip the `idl/` from `foxsi_root_path` and replace it with `data/`.  Your version doesn't work because `data/` is not a subdirectory of `idl/`.

**Comment by aringlis** [comment]
Oh I see, you have to explicitly be in the foxsi-smex/idl directory. I reverted the change.

**Comment by ayshih** [comment]
:+1:

**Comment by LinErinG** [comment]
@aringlis  foxsi_get_output_image_cube.pro is breaking for me at line 347 with the new PSF.  (I double-checked; it works ok in the current main version.)  Looking at it, I don't see obviously why it's breaking, but it does look like foxsi_get_output_image_cube.pro doesn't have all the same fixes to it that foxsi_get_output_2d_image.pro has.  Should the same changes be implemented in that version, which does the same thing as 2d_image except repeats it several times for different energies?

**Comment by ehsteve** [comment]
Merging this because Natasha is asking about it. We can do a separate bug fix for @LinErinG issue.

**Comment by aringlis** [comment]
@LinErinG my mistake - I neglected the image_cube file. I'll post a fix for this.

**Comment by ayshih** [comment]
@aringlis sorry, was already working on it, and just posted the PR (#24)

**Comment by aringlis** [comment]
@ayshih - awesome, thanks!

**Comment by LinErinG** [comment]
@aringlis @ehsteve @ayshih Reviving an old, finished pull request.  This function doesn't seem to work for me.  I get the same PSF no matter what off-axis position I put into it.  @aringlis can you verify that this function returned significantly different PSFs for different off-axis positions at the time you did the PR?  Does it work when you try it now?

**Comment by ayshih** [comment]
@LinErinG Regardless of whether or not the code is working as intended, the current PSF parameters file has no dependence on off-axis angle (as of commit 9782cb96f2f8a315d97d46124f61bfe6fb8990e9), so you would see no variation with off-axis angle without changing that file.

**Comment by LinErinG** [comment]
@aringlis @ehsteve @ayshih Ok, I'll add this as an issue, since one can only find that out by digging in the code to see what it calls.  Does a parameters file for off-axis angles exist in anybody's branch?  We need it for a FOXSI rocket application, and there's time pressure.

**Comment by aringlis** [comment]
@LinErinG @ayshih @ehsteve - I have an old psf_parameters.txt file that seems to have information as a function of off-axis angle. But, I dimly recall that we intentionally removed this functionality from the code, as we were not convinced we were doing it properly - does that ring any bells?

**Comment by ayshih** [comment]
That commit I referenced (9782cb9) has that old parameters file, but @ehsteve will need to refresh everyone's memories on the precise reasons for zeroing out the dependence.
